### DISCUSSION TITLE
What integer size is used for length delimiter?

### DISCUSSION THREAD

**Discussion by matthewmturner** [tokio-rs/prost#543] [General]
When using `encode_length_delimited` or `decode_length_delimited` what integer size is used to represent the length?  I have mutiple protocol buf messages in a file and i want to iterate through and read them but i think i need to allocate a buffer of the right size in order to do this - which would require knowing the size of the length delimiter.

Unless i'm doing something wrong?

Thanks!

**Comment by fd** [comment]
As per the [documentation](https://docs.rs/prost/0.8.0/prost/fn.decode_length_delimiter.html) you should expect a length delimiter to be between 1 and 10 bytes. It is encoded as a variable length `uint64`. The easiest way to read a length delimited stream is to uses a buffered stream and and peek the first 10 bytes from the stream to decode the length. then use [`prost::length_delimiter_len`](https://docs.rs/prost/0.8.0/prost/fn.length_delimiter_len.html) to know the number of bytes to skip.

**Reply by matthewmturner** [reply]
@fd thank you for the response!  im not sure how i missed that going through the documents.  do you have any recommendations for working with variable length integers in rust?

appreciate the help!

**Reply by mumbleskates** [reply]
turns out you can also just pass `&mut impl Buf` anyway, instead of by-value, because there's a blanket implementation for that in the `Bytes` crate. so the by-val signature makes mutating the original buf optional, rather than mandatory.
